### DISCUSSION TITLE
createManyAndReturn() return order

### DISCUSSION THREAD

**Discussion by smithy747** [prisma/prisma#24894] [Q&A] [answered]
### Bug description

I looked at the Prisma Client API Reference and noticed there wasn't any information on the order of the entries returned from the createManyAndReturn() function. This could be potentially useful if you are trying to map some client side data with the data produced from supabase, such as a UUID. Does the data returned map index for index to the data passed in?

For instance, suppose I have a schema:
```prisma
model favorite_colors {
  id             String @id @default(cuid())
  name           String
  favorite_color String
}
```

I have some data: 
```js
newData = [ {name: "John", favorite_color: "Blue"}, {name: "Alex, favorite_color: "Red"}]
```
I make my prisma call:

```
const res = prisma.favorite_colors.createManyAndReturn({
         data: newData
})
```
Assuming a successful batch insert, am I guaranteed that res is:

```js
res = [ 
// First
{
   did: "0190ed7f-f4d2-7b57-aef0-602f4dbf5060",
   name: "John",
   favorite_color: "Blue"
},
// Second
{
  did: "0190ed81-470e-7c17-b2e2-f0ae10f298d3",
  name: "Alex",
  favorite_color: "Red"
}
]
```
or is the following possible?
```js
res = [ 
// First
{
  did: "0190ed81-470e-7c17-b2e2-f0ae10f298d3",
  name: "Alex",
  favorite_color: "Red"
},
// Second
{
   did: "0190ed7f-f4d2-7b57-aef0-602f4dbf5060",
   name: "John",
   favorite_color: "Blue"
},

]
```
I think it would be a good idea to clarify this in the docs.

### How to reproduce

See above

### Prisma information

See above

### Environment & setup

See above

### Prisma Version

See above

**Comment by ludralph** [comment, marked as answer]
Hi @smithy747 👋 

The answer is no, because databases do not explicitly guarantee that themselves.  The order of the returned data from createManyAndReturn() is not guaranteed to match the order of the input data.

**Reply by smithy747** [reply]
Thank you!

**Reply by parimal-donga** [reply]
on that note, do you know what is probability of returning data not-in-order ?
